src/Squirrel.CommandLine/OSX/PackCommand.cs

```diff
@@ -0,0 +1,188 @@
+            ReleaseNotes.ExistingOnly();
+            Add(ReleaseNotes);
+
+            SquirrelAwareExecutable = new Option<string>(new[] { "-e", "--mainExe" }, "The file {NAME} of the main executable") {
+                ArgumentHelpName = "NAME"
+            };
+            Add(SquirrelAwareExecutable);
+
+            Icon = new Option<FileInfo>(new[] { "-i", "--icon" }, "{PATH} to .ico for Setup.exe and Update.exe") {
+                ArgumentHelpName = "PATH"
+            };
+            Icon.ExistingOnly().RequiresExtension(".ico");
+            Add(Icon);
+
+            BundleId = new Option<string>("--bundleId", "Override the apple unique {ID} when generating bundles") {
+                ArgumentHelpName = "ID"
+            };
+            Add(BundleId);
+
+            NoDelta = new Option<bool>("--noDelta", "Skip the generation of delta packages");
+            Add(NoDelta);
+
+            NoPackage = new Option<bool>("--noPkg", "Skip generating a .pkg installer");
+            Add(NoPackage);
+
+            //TODO: Would be nice to setup completions at least for the keys of this option
+            PackageContent = new Option<KeyValuePair<string, FileInfo>[]>("--pkgContent", (ArgumentResult value) => {
+                var splitCharacters = new[] { '=', ':' };
+                var results = new List<KeyValuePair<string, FileInfo>>();
+                for (int i = 0; i < value.Tokens.Count; i++) {
+                    string token = value.Tokens[i].Value;
+                    string[] parts = token.Split(splitCharacters, 2);
+                    switch(parts.Length) {
+                        case 1:
+                            results.Add(new KeyValuePair<string, FileInfo>(parts[0], new FileInfo("")));
+                            break;
+                        case 2:
+                            results.Add(new KeyValuePair<string, FileInfo>(parts[0], new FileInfo(parts[1])));
+                            break;
+                    }
+                }
+                return results.ToArray();
+            }, false, "Add content files (eg. readme, license) to pkg installer.");
+            PackageContent.AddValidator((OptionResult result) => {
+                var validContentKeys = new HashSet<string> {
+                    "welcome",
+                    "readme",
+                    "license",
+                    "conclusion",
+                };
+                foreach (var kvp in result.GetValueForOption(PackageContent)) {
+                    if (!validContentKeys.Contains(kvp.Key)) {
+                        result.ErrorMessage = $"Invalid {PackageContent.Name} key: {kvp.Key}. Must be one of: " + string.Join(", ", validContentKeys);
+                    }
+
+                    if (!kvp.Value.Exists) {
+                        result.ErrorMessage = $"{PackageContent.Name} file not found: {kvp.Value}";
+                    }
+                }
+            });
+            PackageContent.ArgumentHelpName = "key=<FILE>";
```

Comment:
Is this a reasonable way to display to the user how to specify the package content?

Reply:
In Mono.Options, the way you specify this was `--pkgContent welcome mywelcomefile.html`. Assuming an `=` is now required I guess this is fine.

Reply:
It could be made to work that way, but Mono.Options relies much more on the position of the raw arguments, while System.CommandLine allows for specifying things in any reasonable order, and matching tokens up with the relevant options/commands. We could make the original format work, but it gets complicated because of the space, terminals will treat those as separate tokens. Having something like an `=` or `:` will link the values together. Alternatively we could always allow for spaces between them if the caller knew to quote the argument value (ie. `--pkgContent "welcome mywelcomefile.html"`)

Reply:
How would this new parsing handle paths with quotes and spaces? eg. `--pkgContent welcome="folder name/mywelcomefile.html"` ?